pkg/kubelet/kubelet_pods.go

```diff
@@ -2095,18 +2095,14 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 				}
 			}
 		}
-		container := kubecontainer.GetContainerSpec(pod, cName)
 
 		// Always set the status to the latest allocated resources, even if it differs from the
 		// allocation used by the current sync loop.
 		alloc, found := kl.statusManager.GetContainerResourceAllocation(string(pod.UID), cName)
-		if found {
-			status.AllocatedResources = alloc.Requests
-		} else if !(container.Resources.Requests == nil && container.Resources.Limits == nil) {
-			// This case is expected for ephemeral containers.
-			if oldStatusFound {
-				status.AllocatedResources = oldStatus.AllocatedResources
```

Comment:
Just to verify my understanding, the original code here copies the allocated resources from the previous status. We don't need this anymore because `GetContainerResourceAllocation` should always return the latest status?

Reply:
Correct. I can't think of any reason the old status would ever have AllocatedResources set if the resources aren't in the allocated resources cache.